### PR TITLE
ads: move mutex to proxy

### PR DESCRIPTION
**Pull request by ramaraochavali**
This PR
- Moves the `mutex` to `Proxy` from `Connection` where it is only used for protecting proxy changes
- Rename `node` -> `proxy` in `Connection`
- Minor comment fixes

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/xds/ads.go [hunk 1]
```diff
@@ -301,16 +304,6 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream discovery.AggregatedD
...
-			// It is called when config changes.
-			// This is not optimized yet - we should detect what changed based on event and only
-			// push resources that need to be pushed.
-
-			// TODO: possible race condition: if a config change happens while the envoy
-			// was getting the initial config, between LDS and RDS, the push will miss the
-			// monitored 'routes'. Same for CDS/EDS interval.
-			// It is very tricky to handle due to the protocol - but the periodic push recovers
-			// from it.
-
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ads.go:313]: I donot think this has been resolved

Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go:313]: L308-312

Comment (ramaraochavali) [reply pilot/pkg/xds/ads.go:313]: Ok. Reverted that TODO

**Pull request merged**